src/huggingface_hub/repository.py

```diff
@@ -226,6 +226,31 @@ def is_git_ignored(filename: Union[str, Path]) -> bool:
     return is_ignored
 
 
+def is_binary_file(filename: Union[str, Path]) -> bool:
+    """
+    Check if file is a binary file.
+
+    Args:
+        filename (`str` or `Path`):
+            The filename to check.
+
+    Returns:
+        `bool`: `True` if the file passed is a binary file, `False` otherwise.
+    """
+    try:
+        with open(filename, "rb") as f:
+            content = f.read()
```

Comment:
```suggestion
            content = f.read(1024)  # or 512 if we want to be consistent with the backend
```

Reply:
This was discussed above here: https://github.com/huggingface/huggingface_hub/pull/828#discussion_r848732391

Do you disagree with the conclusion?

Reply:
The solution currently implemented loads a maximum of 10MB in memory when calling `git_add`: it  tracks [large files](https://github.com/huggingface/huggingface_hub/blob/9fe6bb6f303d0b642bfed753866bf5aab1378376/src/huggingface_hub/repository.py#L1167) before tracking [binary files](https://github.com/huggingface/huggingface_hub/blob/9fe6bb6f303d0b642bfed753866bf5aab1378376/src/huggingface_hub/repository.py#L1170).

When tracking binary files, it looks at files [which are not yet tracked with lfs](https://github.com/huggingface/huggingface_hub/blob/9fe6bb6f303d0b642bfed753866bf5aab1378376/src/huggingface_hub/repository.py#L1031), eliminating large files.

Instead of the 1MB limit that you propose here, we could instead put a max of 10MB [here](https://github.com/huggingface/huggingface_hub/blob/9fe6bb6f303d0b642bfed753866bf5aab1378376/src/huggingface_hub/repository.py#L242), which will only be triggered when `auto_track_binary_files` is called independently of `git_add` (which is a possibility!).

Reply:
yes, but you also want to have this method public, which means people can call it before having tracked large files.

I'm happy with your suggestion of 10MB.

Reply:
Sounds good, thanks for your review. Addressed in [cbfdce5](https://github.com/huggingface/huggingface_hub/pull/828/commits/cbfdce5d3c5af3cfefc493e2111c63e382cb2f3d)